sig-scheduling/CONTRIBUTING.md

```diff
@@ -72,6 +72,26 @@ to the issue/PR, which helps to preserve the context.
 
 * Always open an issue for a TODO or a follow-up just in case you forget it.
 
+### Technical and style guidelines
+
+The following guidelines apply primarily to kube-scheduler, but some subprojects
+might also adhere to them.
+
+- Follow [effective go](https://go.dev/doc/effective_go) guidelines.
+- Naming:
+  - Length: As a rule-of-thumb, the length of a variable name should be
+    proportional to the size of the scope where it is used and inversely
+    proportional to the number of times that it is used.
+- In tests:
+  - Follow a [DAMP principle](https://stackoverflow.com/a/11837973).
+  - Use `cmp.Diff` instead of `reflect.Equal`, to provide useful comparisons.
+  - Avoid creating or using assertion libraries.
+    Use standard `t.Error` or `t.Fatal`, as necessary.
+
```

Comment:
I think we may want to mention that in tests, comparing the error using `errors.Is()` (`cmpopts.EquateErrors()`) instead of comparing the error message if it is possible.

WDYT?

Reply:
Yes :)

Reply:
The "if it is possible" didn't make it into the merged text.

@tenzen-y: you were probably thinking of packages that define certain errors as part of their API and then testing that really those errors are returned when they should be returned, right? That is what callers of the package check for via `errors.Is`.

If a package has no defined errors, then a unit test can only check that some error is returned when a failure is expected.

Reply:
Then the question should be why is the error not defined.

But I would be ok having a catch-all error that can be used as value for `wantErr` when you really don't care.